src/libOpenImageIO/imagebufalgo.cpp

```diff
@@ -1098,7 +1084,7 @@ ImageBufAlgo::fillholes_pushpull (ImageBuf &dst, const ImageBuf &src,
 
     // Now copy the completed base layer of the pyramid back to the
     // original requested output.
-    paste (dst, dstspec.x, dstspec.y, dstspec.z, 0, *pyramid[0]);
+    paste (dst, src.spec().x, src.spec().y, src.spec().z, 0, *pyramid[0]);
```

Comment:
This line is the actual bug fix. The rest is just cleanup, testing, and docs clarification.